pkg/completer/completer.go

```diff
@@ -0,0 +1,53 @@
+package completer
+
+import (
+	"strings"
+
+	"github.com/calyptia/cli/pkg/helpers"
+	fluentbitconfig "github.com/calyptia/go-fluentbit-config"
+	"github.com/spf13/cobra"
+)
+
+type Completer struct {
+}
+
+func (c *Completer) CompletePluginKinds(cmd *cobra.Command, args []string, toComplete string) ([]string, cobra.ShellCompDirective) {
```

Comment:
Instead of having a single `Completer` with a receiver for each object we really need a base interface and an implementation for each. Overloading it all into a single `struct` doesn't seem very modular to me.

For kinds, output formats and (maybe) plugin names it might be OK, but not so much for pipelines, core_instances, etc... which should maybe have their specific logic in their own model. The big risk their might be recursive imports though... :thinking:

Reply:
I haven't yet got to the point of deep diving into logic of each of those "completers". One step at the time. 

I'm trying to make PRs as small as possible.